extensions/amp-viewer-assistance/0.1/amp-viewer-assistance.js

```diff
@@ -70,14 +70,18 @@ export class AmpViewerAssistance {
    */
   actionHandler_(invocation) {
     const {method, args} = invocation;
-    if (method == 'updateActionState' && !!args) {
-      this.viewer_./*OK*/sendMessageAwaitResponse(method, args).catch(error => {
-        user().error(TAG, error.toString());
-      });
-    } else if (method == 'signIn') {
-      this.requestSignIn_();
+    if (method === 'updateActionState') {
+      // "updateActionState" requires a low-trust event.
+      if (invocation.satisfiesTrust(ActionTrust.LOW) && args) {
```

Comment:
@hellokoji Are we sure we want to allow arbitrary arguments to `updateActionState`? It's more error prone due to typos for example.

Reply:
Currently, `updateActionState` takes an object parameter with an `actionStatus` string field and a `result` object field. We could make a white-list on the `actionStatus` string to error out quicker given an invalid argument, but for the `result` field, the schema can get pretty complex.

Right now we support `COMPLETED_ACTION_STATUS`, `ACTIVE_ACTION_STATUS`, and `FAILED_ACTION_STATUS`. I am okay with white-listing just those three.